src/hotspot/share/utilities/bitMap.cpp

```diff
@@ -99,6 +99,70 @@ void GrowableBitMap<BitMapWithAllocator>::resize(idx_t new_size_in_bits, bool cl
   update(map, new_size_in_bits);
 }
 
+template <class BitMapWithAllocator>
+bm_word_t* GrowableBitMap<BitMapWithAllocator>::copy_of_range(idx_t start_bit, idx_t end_bit, bool clear) {
+  assert(start_bit < end_bit, "End bit must come after start bit");
+  assert(end_bit <= size(), "End bit not in bitmap");
+
+  // We might have extra bits at the end that we don't want to lose
+  idx_t cutoff = bit_in_word(end_bit);
+  idx_t start_word = to_words_align_down(start_bit);
+  idx_t end_word = to_words_align_up(end_bit);
+  bm_word_t* const old_map = map();
+
+  BitMapWithAllocator* derived = static_cast<BitMapWithAllocator*>(this);
+
+  bm_word_t* new_map = derived->allocate(end_word - start_word);
+
+  // All words need to be shifted by this amount
+  idx_t shift = bit_in_word(start_bit);
+  // Bits shifted out by a word need to be passed into the next
+  idx_t carry = 0;
+
+  // Iterate the map backwards as the shift will result in carry-out bits
+  for (idx_t i = end_word; i --> start_word;) {
+    // First iteration is a special case:
+    // There may be left over bits in the last word that we want to keep while discarding the rest
+    if (i == end_word - 1 && cutoff > 0) {
```

Comment:
I suppose it isn't necessary to clear the leftover MSB so that detail can be excluded. The junk bits will still be present however and that may introduce some confusion when debugging.